contrib/IECoreAppleseed/src/IECoreAppleseed/outputDriver/DisplayTileCallback.cpp

```diff
@@ -226,131 +240,100 @@ class DisplayLayer
 			}
 		}
 
-		Box2i box_inside_data_window( int x, int y, int w, int h ) const
+		Box2i boxInsideDataWindow( int x, int y, int w, int h ) const
 		{
 			int x1 = x + w - 1;
 			int y1 = y + h - 1;
 
-			return Box2i( V2i( std::max( x , m_data_window.min.x ), std::max( y , m_data_window.min.y ) ),
-						  V2i( std::min( x1, m_data_window.max.x ), std::min( y1, m_data_window.max.y ) ) );
+			return Box2i( V2i( std::max( x , m_dataWindow.min.x ), std::max( y , m_dataWindow.min.y ) ),
+						  V2i( std::min( x1, m_dataWindow.max.x ), std::min( y1, m_dataWindow.max.y ) ) );
 		}
 
+		using Mutex = std::mutex;
+		using LockGuard = std::lock_guard<Mutex>;
+
+		// Our public methods will be called concurrently,
+		// so we use this mutex to protect access to our
+		// members.
+		Mutex m_mutex;
 		DisplayDriverPtr m_driver;
 		asf::Image *m_image;
-		Box2i m_data_window;
+		Box2i m_dataWindow;
 		vector<float> m_buffer;
-		size_t m_tile_width;
-		size_t m_tile_height;
-		size_t m_channel_count;
+		size_t m_tileWidth;
+		size_t m_tileHeight;
+		size_t m_channelCount;
 		string m_layerName;
 		asf::Dictionary m_params;
 
 };
 
+using DisplayLayers = vector<unique_ptr<DisplayLayer>>;
+using DisplayLayersPtr = shared_ptr<DisplayLayers>;
+
 class DisplayTileCallback : public ProgressTileCallback
 {
 
 	public :
 
-		explicit DisplayTileCallback( const asr::ParamArray &params ) : ProgressTileCallback(), m_displays_initialized( false )
+		DisplayTileCallback( const DisplayLayersPtr &layers )
+			:	m_layers( layers )
 		{
-			// Create display layers.
-			m_layers.reserve( params.dictionaries().size() );
-			for( auto it( params.dictionaries().begin() ), eIt( params.dictionaries().end() ); it != eIt; ++it )
-			{
-				m_layers.push_back( new DisplayLayer( it.key(), it.value() ) );
-			}
 		}
 
 		~DisplayTileCallback() override
 		{
-			for( DisplayLayer *layer : m_layers )
-			{
-				delete layer;
-			}
 		}
 
 		void release() override
 		{
-			// We don't need to do anything here.
-			// The tile callback factory deletes this instance.
+			delete this;
 		}
 
 		void on_tile_begin(const asr::Frame *frame, const size_t tileX, const size_t tileY) override
 		{
 			const asf::CanvasProperties &props = frame->image().properties();
 			const size_t x = tileX * props.m_tile_width;
 			const size_t y = tileY * props.m_tile_height;
-
-			boost::lock_guard<boost::mutex> guard( m_mutex );
-
-			if( m_displays_initialized )
+			for( auto &layer : *m_layers )
```

Comment:
We can't copy, because `layer` is a `unique_ptr`. I've moved everything to use `const auto &layer` instead, to make it clear that we don't modify the pointer itself.